### PR TITLE
[#944] Add originality threshold flag

### PR DESCRIPTION
**Pull request by SkyBlaise99**
Part of #944

### Proposed commit message

```
The existing setup employs a static originality threshold of 0.51.
However, this threshold is tailored for codes, such as Java or Markdown,
and might not be suitable for other programming languages. Additionally,
it doesn't offer flexibility for users who may want a stricter threshold
but are willing to endure longer processing times, or those who prefer a
more lenient threshold but prioritize faster analysis speeds.

Let's enable users to input their preferred originality threshold.
```

### CODE REVIEW

**Comment by ckcherry23** [comment]
Hi @SkyBlaise99, would it be possible to merge/rebase with `master` so that the CI passes?

**Comment by SkyBlaise99** [comment]
I think the backend side of the main branch was updated, the new introduction of builder pattern is quite a lot of change. I plan to make it a separate PR after these feature prs are merged.

File: src/main/java/reposense/parser/OriginalityThresholdArgumentType.java
Comment (ckcherry23) [inline src/main/java/reposense/parser/OriginalityThresholdArgumentType.java:1]: Is it possible to add tests for this? The codecov diff hit seems low.

**Review by MarcusTXK — APPROVED** [review]
LGTM, great job. I agree that resolving the merge conflict can be done later, especially as more changes to `CliArguments.java` are expected.

It would be great if you could address @ckcherry23 's comment for providing coverage of `OriginalityThresholdArgumentType.java` for lines 17, 20 and 23, but I think if its urgent to merge into #945, you could also do the testing there.

**Review by gok99 — APPROVED** [review]
@SkyBlaise99 So sorry for the review delays!

Haven't taken a deep look at the full analyze-authorship changes, but within the scope PR's scope, changes are looking good to me. On that note, it would be great if you could come up with a simple reviewer guide (explaining key design decisions, changes to the overall architecture, etc) for facilitating the review of  `944-analyze-authorship` given the size of its scope.

Will mark this PR to be merged for now. Let us know if you intent to add tests to this PR before we merge it. If there is a particular hurry to get this merged, I think it's also fine to add the tests in the bigger PR.

**Comment by SkyBlaise99** [comment]
> On that note, it would be great if you could come up with a simple reviewer guide (explaining key design decisions, changes to the overall architecture, etc) for facilitating the review of  `944-analyze-authorship` given the size of its scope.

Imo there isn't actually key design decisions or changes to overall architecture. This is more like a add-on feature to improve the existing code authorship attribution. Nevertheless, I'll provide a rough explanation as to what this code does when the final PR is made. 

> Will mark this PR to be merged for now. Let us know if you intent to add tests to this PR before we merge it.

I can add in the test in this PR to make it more complete. It should be a minor update.

File: src/main/java/reposense/parser/OriginalityThresholdArgumentType.java
Comment (SkyBlaise99) [reply src/main/java/reposense/parser/OriginalityThresholdArgumentType.java:1]: Sure and done

**Comment by gok99** [comment]
Merging this and #2108 ahead of the 2 day window since we're merging to an intermediate branch.

**Pull request merged**